aten/src/ATen/native/Convolution.cpp

```diff
@@ -848,11 +847,9 @@ ConvBackend select_conv_backend(
 
   // don't send empty inputs through backends
   if (input.size(0) == 0 || input.size(1) == 0) {
-    if (input.is_mkldnn() && weight.is_mkldnn()) {
-      return ConvBackend::MkldnnEmpty;
-    } else {
-      return ConvBackend::Empty;
-    }
+    return input.is_mkldnn() ? ConvBackend::MkldnnEmpty : ConvBackend::Empty;
```

Comment:
what about the weight check?

Reply:
I purposefully dropped it because I think it's unnecessary and slightly incorrect. I think we want to return an empty tensor that matches the input (e.g. is a mkldnn tensor) regardless of whether the weight is mkldnn or not - do you agree?

Reply:
That sounds good :)
For future ones, we can split such "dangerous" change to a different PR to be able to easily revert if it breaks something unexpected.